### ISSUE TITLE
Allow `mnd.ignored-numbers`  validation works with integer

### ISSUE THREAD

**Issue by manuelarte** [golangci/golangci-lint#6322]
### Welcome

- [x] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [x] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### How did you install golangci-lint?

Brew

### Your feature request related to a problem? Please describe

When running:

```bash
golangci-lint config verify
```

in the followning golangci conf:

```yaml
---
version: "2"

linters:
  enable:
    - mnd
  settings:
    mnd:
      ignored-numbers:
        - 0

```

I got the error:

```
jsonschema: "linters.settings.mnd.ignored-numbers.0" does not validate with "/properties/linters/properties/settings/properties/mnd/$ref/properties/ignored-numbers/items/type": got number, want string
The command is terminated due to an error: the configuration contains invalid elements
```

I would expect that `ignored-numbers` accept an integer also as a valid value.

### Describe the solution you'd like

Maybe I am too naive, but just by changing the json schema to:

```json
"ignored-numbers": {
  "description": "List of numbers to exclude from analysis.",
  "examples": [["1000", "1234_567_890", "3.14159264"]],
  "type": "array",
  "items": {
    "type": ["string", "integer"]
  }
},
```

Now `ignored-numbers` support `integer` values. Tested here: https://github.com/manuelarte/golangci-lint/tree/enable-integer-settings-mnd-ignored-numbers

If desired, I can create a PR.

### Describe alternatives you've considered

The alternative is just to set the numbers as strings.

### Supporter

- [ ] I am a sponsor through [GitHub](https://github.com/sponsors/golangci/) or [OpenCollective](https://opencollective.com/golangci-lint)

**Comment by ldez** [comment]
Internally, `mnd` used `string` and not numbers.

Also, the values can be `int` but also `float`, and we all know the problem with floats.
There are the same kind of problems with octal.

So we will keep `string`.

**Comment by ldez** [comment]
🤔 as it's only a JSONSchema modification, it can work but `integer` is not the right type.

**Comment by ldez** [comment]
I think `number` is the right type.

**Comment by ldez** [comment]
🤔 it creates side-effects when using octal (ex: `0777`).
